tests/phpunit/tests/media.php

```diff
@@ -3948,6 +3948,105 @@ public function data_special_contexts_for_the_content() {
 		);
 	}
 
+	/**
+	 * Tests that the content media count is not affected by `the_excerpt()` calls for posts that contain images.
+	 *
+	 * @ticket 56588
+	 *
+	 * @covers ::the_excerpt
+	 * @covers ::wp_trim_excerpt
```

Comment:
While this test makes sure these two functions aren't triggering side effects, we're not really testing those functions, but are instead testing `wp_increase_content_media_count()`,  so maybe we should update this to the following?:

```suggestion
	 * @covers :: wp_increase_content_media_count
```

Reply:
@joemcgill Here I disagree. The `wp_increase_content_media_count()` function was not touched as part of this PR, so nothing here is testing that function. It's testing how the function is called, which is happening in `wp_trim_excerpt()`, hooked into `the_excerpt` filter.

Reply:
My thinking here is that `@covers` is meant to specify [which functions should be shown in coverage reports as being covered by unit tests](https://docs.phpunit.de/en/9.5/annotations.html#covers) and filter out others used in the method, not just to specify which functions are being called within the test method.

This one is a tricky one because we're trying to test that calling `wp_trim_excerpt ` in specific cases doesn't cause side effects by increasing the content media count unnecessarily, but we're not really testing the output of `the_excerpt` or `wp_trim_excerpt`. Perhaps `wp_trim_excerpt` is correct and using that alone would be enough?

Reply:
But `wp_increase_content_media_count()` is not covered by this test, even though it's used. This test covers a logic change in `wp_trim_excerpt()`, which doesn't change the output of that function, but unit tests aren't limited to what a function outputs. FWIW, the fact that this function has side effects isn't great, but that's WordPress :)

I agree `the_excerpt` should probably be removed, since indeed the logic change tested here is in `wp_trim_excerpt()`, but not in `the_excerpt()`. I'll update.

Reply:
Updated in https://github.com/WordPress/wordpress-develop/pull/4439/commits/d613c309bb7aa18089c38571a39da055f6af41e7